### PR TITLE
W504 - line break after binary operator

### LINKED ISSUE #1659: [Travis] flake8 was updated to 3.6.0
flake8 tests are broken thanks to 'W504 line break after binary operator' and ' W605 invalid escape sequence'.

See https://travis-ci.org/secdev/scapy/jobs/445176377

### PR DESCRIPTION
**Pull request by guedou**
This PR fixes #1659. I tried to limit the use of `noqa` but it is needed in some places.

### CODE REVIEW

**Comment by guedou** [comment]
@gpotter2 @p-l- ready to be reviewed!

File: scapy/arch/bpf/core.py [hunk 1]
```diff
@@ -70,9 +70,9 @@ def get_if_raw_hwaddr(ifname):
         raise Scapy_Exception("Failed to execute ifconfig: (%s)" % msg)
 
     # Get MAC addresses
-    addresses = [l for l in fd.readlines() if l.find("ether") >= 0 or
-                 l.find("lladdr") >= 0 or
-                 l.find("address") >= 0]
+    filter_line = lambda l: l.find("ether") >= 0 or l.find("lladdr") >= 0 or \
+        l.find("address") >= 0
+    addresses = [l for l in fd.readlines() if filter_line(l)]
```
Comment (p-l-) [inline scapy/arch/bpf/core.py]: On this one, I'd rather keep the old code, which looks more readable to me.

Comment (guedou) [reply scapy/arch/bpf/core.py]: Done.
I had to use `noqa`.

File: scapy/layers/dns.py [hunk 2]
```diff
@@ -72,8 +72,9 @@ def dns_get_str(s, p, pkt=None, _internal=False):
                 warning("DNS decompression loop detected")
                 break
             elif not _internal:
-                raise Scapy_Exception("DNS message can't be compressed" +
-                                      "at this point!")
+                msg = "DNS message can't be compressed"
+                msg += "at this point!"
+                raise Scapy_Exception(msg)
```
[the remark this answers is not in the document]
Comment (guedou) [reply scapy/layers/dns.py]: Done.

File: scapy/layers/dot15d4.py [hunk 3]
```diff
@@ -238,9 +238,8 @@ def guess_payload_class(self, payload):
                        "802.15.4 protocol. Values must be in the list: "
             else:
                 _msg = "Unknown conf.dot15d4_protocol value: must be in "
-            warning(_msg +
-                    "['sixlowpan', 'zigbee']" +
-                    " Defaulting to SixLoWPAN")
+            _msg += "['sixlowpan', 'zigbee']" + " Defaulting to SixLoWPAN"
```
[the remark this answers is not in the document]
Comment (guedou) [reply scapy/layers/dot15d4.py]: Done

**Comment by p-l-** [comment]
(copy & paste of my review message here)

Is it worth it? Maybe we could decide we ignore W504 globally?

**Comment by guedou** [comment]
@p-l- most of the W504 errors were easily fixed. It was a lot of work because the code has many of them, but it seems OK to enforce W504.

**Comment by gpotter2** [comment]
I don’t see tests failing as abruptly:/ I’ll try to figure it out

**Comment by guedou** [comment]
Is it also failing on a real Windows system?

File: scapy/arch/bpf/core.py [hunk 4]
```diff
@@ -70,8 +70,8 @@ def get_if_raw_hwaddr(ifname):
         raise Scapy_Exception("Failed to execute ifconfig: (%s)" % msg)
 
     # Get MAC addresses
-    addresses = [l for l in fd.readlines() if l.find("ether") >= 0 or
-                 l.find("lladdr") >= 0 or
+    addresses = [l for l in fd.readlines() if l.find("ether") >= 0 or  # noqa: W504, E501
```
Comment (p-l-) [inline scapy/arch/bpf/core.py]: Could you move the `if` to a next line and remove at least the `E501` new `noqa`?

File: scapy/arch/windows/__init__.py [hunk 5]
```diff
@@ -868,10 +870,10 @@ def _ask_user():
                         self.load_from_powershell()
                         return
                 _error_msg = "Could not start the pcap service ! "
-            warning(_error_msg +
-                    "You probably won't be able to send packets. "
-                    "Deactivating unneeded interfaces and restarting Scapy might help. "  # noqa: E501
-                    "Check your winpcap and powershell installation, and access rights.")  # noqa: E501
+            _error_msg += "You probably won't be able to send packets. "
```
Comment (p-l-) [inline scapy/arch/windows/__init__.py]: I don't understand this change. Wouldn't it be cleaner to keep the previous version and cut the longest lines to remove the `noqa`?

Comment (guedou) [reply scapy/arch/windows/__init__.py]: noqa removed

File: scapy/arch/bpf/core.py (hunk 4 above)
Comment (guedou) [reply scapy/arch/bpf/core.py]: rewritten with `in` instead of `find()` to save space.

**Comment by p-l-** [comment]
> @p-l- most of the W504 errors were easily fixed. It was a lot of work because the code has many of them, but it seems OK to enforce W504.

I'm not sure to agree. IMO W504 should be permitted. They are disabled by default, and I'm not sure it makes sens to forbid both W503 and W504.

**Comment by guedou** [comment]
I agree, enforcing both don’t make any sense. I did not notice that before your comment.

As this PR does not work, it is safe to close it and decide to only enforce W503.

**Pull request closed without merging**